x-pack/plugins/stack_alerts/public/alert_types/search_threshold/expression.tsx

```diff
@@ -0,0 +1,139 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiSpacer, EuiCallOut, EuiTitle } from '@elastic/eui';
+import {
+  COMPARATORS,
+  ThresholdExpression,
+  ForLastExpression,
+  AlertTypeParamsExpressionProps,
+} from '../../../../triggers_actions_ui/public';
+import { SearchThresholdAlertParams } from './types';
+
+export const DEFAULT_VALUES = {
+  THRESHOLD_COMPARATOR: COMPARATORS.GREATER_THAN,
+  TIME_WINDOW_SIZE: 5,
+  TIME_WINDOW_UNIT: 'm',
+  THRESHOLD: [1000],
+};
+
+const expressionFieldsWithValidation = ['threshold0', 'threshold1', 'timeWindowSize'];
+
+export const SearchThresholdAlertTypeExpression: React.FunctionComponent<
+  AlertTypeParamsExpressionProps<SearchThresholdAlertParams>
+> = ({ alertParams, setAlertParams, setAlertProperty, errors }) => {
+  const { thresholdComparator, threshold, timeWindowSize, timeWindowUnit, searchSource } =
+    alertParams;
+
+  const hasExpressionErrors = !!Object.keys(errors).find(
+    (errorKey) =>
+      expressionFieldsWithValidation.includes(errorKey) &&
+      errors[errorKey].length >= 1 &&
+      alertParams[errorKey as keyof SearchThresholdAlertParams] !== undefined
+  );
+
+  const expressionErrorMessage = i18n.translate(
+    'xpack.stackAlerts.searchThreshold.ui.alertParams.fixErrorInExpressionBelowValidationMessage',
+    {
+      defaultMessage: 'Expression contains errors.',
+    }
+  );
+
+  const setDefaultExpressionValues = async () => {
+    setAlertProperty('params', {
+      searchSource: searchSource ?? {},
+      threshold: threshold ?? DEFAULT_VALUES.THRESHOLD,
+      thresholdComparator: thresholdComparator ?? DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+      timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+      timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+    });
+  };
+
+  useEffect(() => {
+    if (searchSource) {
+      setDefaultExpressionValues();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!searchSource) {
+    // temporary catch when people try to create the alert in management
+    return (
+      <div>
+        <EuiCallOut
+          color="danger"
+          size="s"
+          title={`Currently the creation of this rule is just possible in Discover`}
+        />
+        <EuiSpacer size="l" />
+      </div>
+    );
+  }
+
+  return (
+    <>
+      {hasExpressionErrors ? (
+        <Fragment>
+          <EuiSpacer />
+          <EuiCallOut color="danger" size="s" title={expressionErrorMessage} />
+          <EuiSpacer />
+        </Fragment>
+      ) : null}
+      <div>
+        Imagine or implement data view, query, filters using search source here. Not editable in the
```

Comment:
i18n.missing?

Reply:
yes, generally, don't think this should make it into the final implementation, so it's more like a comment :)